### ISSUE TITLE
Delete User Functionality

### ISSUE THREAD

**Issue by blchelle** [blchelle/collabogreat#159]
## Description
Give users the ability to delete their account. Upon deletion, the following needs to happen
1. Reassign all of their current tasks to 'unassigned' / null (Blocked by another issue)
2. Delete all their projects if they were the only member of that project
3. Get rid of the user in all the projects where there are other users.
